base/combinatorics.jl

```diff
@@ -276,7 +276,7 @@ julia> B[invperm(v)]
 """
 function invperm(a::AbstractVector)
     require_one_based_indexing(a)
-    b = zero(a) # similar vector of zeros
+    fill!(similar(a), zero(eltype(a))) # similar vector of zeros
     n = length(a)
     @inbounds for (i, j) in enumerate(a)
         ((1 <= j <= n) && b[j] == 0) ||
```

Comment:
```suggestion
        (1 <= j <= n) ||
```

Reply:
This is to make sure `b[j]` haven't been visited before (there is no `isperm(a)`). With your suggestions:
```
julia> invperm([1, 1, 1])
3-element Vector{Int64}:
                   3
 7021782956690391410     #  garbage from similar(...)
                   0     #  garbage from similar(...)
```

Reply:
Ah, I see! I didn't realized these weren't just visited in order, ignore my comment